rero_ils/modules/stats/api/report.py

```diff
@@ -0,0 +1,445 @@
+                    size=self.aggs_size),
+                'owning_library': lambda: A(
+                    'terms', field='loan.item.library_pid',
+                    size=self.aggs_size)
+            },
+            'get_key': {
+                'transaction_location': lambda bucket:
+                f'{self.locations[bucket.key]} ({bucket.key})',
+                'transaction_month': lambda bucket: bucket.key_as_string,
+                'transaction_year': lambda bucket: bucket.key_as_string,
+                'patron_type': lambda bucket: bucket.key,
+                'patron_age': lambda bucket: bucket.key,
+                'document_type': lambda bucket: bucket.key,
+                'patron_postal_code': lambda bucket: bucket.key,
+                'transaction_channel': lambda bucket: bucket.key,
+                'owning_library': lambda bucket:
+                f'{self.libraries[bucket.key]} ({bucket.key})'
+            }
+        }
+
+    def _process_aggregations(self, es_results):
+        """Process the es aggregations structures."""
+        cfg = self.es_config.get(self.indicator)
+        results = {}
+        y_keys = set()
+        iter_distrib = iter(self.distributions)
+        if distrib1 := next(iter_distrib, None):
+            distrib2 = next(iter_distrib, None)
+            get_key1 = cfg['get_key'][distrib1]
+            for dist1 in es_results.aggs[distrib1].buckets:
+                if isinstance(dist1, str):
+                    key1 = dist1
+                    parent_dist = es_results.aggs[distrib1].buckets[key1]
+                    doc_count = \
+                        es_results.aggs[distrib1].buckets[key1].doc_count
+                else:
+                    parent_dist = dist1
+                    key1 = get_key1(dist1)
+                    doc_count = dist1.doc_count
+                if not doc_count:
+                    continue
+                results[key1] = dict(count=doc_count)
+                values = {}
+                if distrib2:
+                    get_key2 = cfg['get_key'][distrib2]
+                    for dist2 in parent_dist[distrib2].buckets:
+                        if isinstance(dist2, str):
+                            key2 = dist2
+                            doc_count = dist1[distrib2].buckets[key2].doc_count
+                        else:
+                            key2 = get_key2(dist2)
+                            doc_count = dist2.doc_count
+                        values[key2] = doc_count
+                        y_keys.add(key2)
+                if values:
+                    results[key1]['values'] = values
+            y_keys = sorted(y_keys)
+            x_keys = sorted(results.keys())
+        return self._process_distributions(x_keys, y_keys, results)
+
+    def _process_distributions(self, x_keys, y_keys, results):
+        """Process the elasticsearch aggregations results."""
+        data = []
+        if y_keys:
+            data.append([''] + y_keys)
```

Comment:
Why create an empty list and adding `y_keys` into it ? Why not simply use `data.extend(y_keys)` ?